packages/sitecore-jss/src/site/site-resolver.ts

```diff
@@ -52,6 +45,34 @@ export class SiteResolver {
     return siteInfo;
   };
 
+  protected getHostMap = (): Map<string, SiteInfo> => {
+    const map = new Map<string, SiteInfo>();
+
+    // First collect unique hostnames.
+    // For sites with same hostname defined, priority is given to the first encountered.
+    this.sites.forEach((site) => {
+      const hostnames = site.hostName
+        .replace(/\s/g, '')
+        .toLocaleLowerCase()
+        .split(DELIMITERS);
+      hostnames.forEach((hostname) => {
+        if (!map.has(hostname)) {
+          map.set(hostname, site);
+        }
+      });
+    });
+
+    // Now order by specificity.
+    // This equivalates to sorting from longest to shortest with the assumption
+    // that your match is less specific as wildcards are introduced.
+    // E.g. order.eu.site.com → *.eu.site.com → *.site.com → *
```

Comment:
One case which I've imagined:
if you have _*.eu.site.com_ and _i.eu.site.com_
both have the same length, but the second one is more specific, in case the length is the same, can we count wildcards, and if there are more in this case it's less specific?

Reply:
Probably an edge case, but worth covering for sure. Done.